collect_app/src/main/java/org/odk/collect/android/logic/PropertyManager.java

```diff
@@ -134,7 +133,7 @@ public List<String> getProperty(String propertyName) {
     @Override
     public String getSingularProperty(String propertyName) {
         if (!permissionsProvider.isReadPhoneStatePermissionGranted() && isPropertyDangerous(propertyName)) {
-            eventBus.post(new ReadPhoneStatePermissionRxEvent());
+            isPhoneNumberRequired.postValue(true);
```

Comment:
Could this be `setValue` or is `getSingularProperty` called from background (non-UI) threads?

Reply:
Yes it's used from background so it needs to be `postValue`